### PR TITLE
Dismiss swipe actions when touching outside the active cell

### PR DESCRIPTION
**Pull request by kyleve**
I realized this behavior was missing: When touching down outside of the actively swiped cell, we should dismiss the active swipe, and let the touch continue to scroll the scroll view.

### Checklist

Please do the following before merging:

- [x] Ensure any public-facing changes are reflected in the [changelog](https://github.com/kyleve/Listable/blob/main/CHANGELOG.md). Include them in the `Main` section.

### CODE REVIEW

**Comment by kylebshr** [comment]
Huh I definitely added this before, it must have gotten removed? https://github.com/kyleve/Listable/pull/130/files#diff-7745c8a7f6e1a90d892771b03ba9a41d35e6ad17de57ad094ee044eb08dc45a4R39

**Comment by kylebshr** [comment]
Removed here: https://github.com/kyleve/Listable/pull/228/files#diff-501c1250e3f754e63a56d3119275705080d224bd3197c74d6f0b2dd70e1f7657L36

In favor of this: https://github.com/kyleve/Listable/pull/228/files#diff-e63bc75982fd7d7cfe6fafc5ac2ddd1507af287f76c4d05716a306c5cbad550dR225

Which is still on main: https://github.com/kyleve/Listable/blob/0f796bb571543b51262bcacf7eff9bfaf5685884/ListableUI/Sources/ListView/ListView.Delegate.swift#L284

But does it not work?

**Comment by kyleve** [comment]
That only works when dragging begins, vs on touch down with no drag - which is what this implements. Eg see this video of Mail.app, each time the swipe dismisses, I'm only touching down, not dragging at all.

https://user-images.githubusercontent.com/327847/182009411-428826fd-4781-4833-851f-50ad8a246b44.MOV

**Comment by kylebshr** [comment]
Got itttt makes sense!

**Comment by kyleve** [comment]
@kylebshr if you have a chance, mind giving this a review?

File: ListableUI/Sources/Internal/ItemCell.swift [hunk 1]
```diff
@@ -179,6 +185,24 @@ final class LiveCells {
...
+            }
+        }
+        
+        return cell?.cell
+    }
+    
+    var activeSwipeCell : AnyItemCell? {
+        first { cell in
+            cell.areSwipeActionsVisible
+        }
```
Comment (n8chur) [inline ListableUI/Sources/Internal/ItemCell.swift:203]: nit:
```suggestion
        first(where: \.areSwipeActionsVisible)
```

File: ListableUI/Sources/ListView/ListView.swift [hunk 2]
```diff
@@ -80,7 +82,14 @@ public final class ListView : UIView, KeyboardObserverDelegate
         self.delegate.layoutManager = self.layoutManager
         
         self.keyboardObserver.add(delegate: self)
-                
+        
+        self.closeActiveSwipesGesture.addTarget(self, action: #selector(closeActiveSwipeGestureIfNeeded))
+        self.addGestureRecognizer(closeActiveSwipesGesture)
```
Comment (n8chur) [inline ListableUI/Sources/ListView/ListView.swift:87]: Can swipe action view use gesture recognizers to observe touch? If they do, is it possible to activate them with this approach or does the `closeActiveSwipesGesture` capture the touch down preventing interaction?

It seems like we may need to ensure that a tap down inside the swipe action space doesn't prevent any gestures from occurring, right?

File: ListableUI/Sources/Internal/ItemCell.swift (hunk 1 above)
Comment (kyleve) [reply ListableUI/Sources/Internal/ItemCell.swift:203]: TIL, I thought this only existed for map!

File: ListableUI/Sources/ListView/ListView.swift (hunk 2 above)
Comment (kyleve) [reply ListableUI/Sources/ListView/ListView.swift:87]: That's what the `shouldRecognize` does – it'll only occur if there is (a) an active swipe, and (b) the touch is not inside the active swipe view

File: ListableUI/Sources/Internal/ItemCell.swift (hunk 1 above)
Comment (n8chur) [reply ListableUI/Sources/Internal/ItemCell.swift:203]: Works with any closure!

Comment (kyleve) [reply ListableUI/Sources/Internal/ItemCell.swift:203]: Magic

File: ListableUI/Sources/ListView/ListView.swift (hunk 2 above)
Comment (n8chur) [reply ListableUI/Sources/ListView/ListView.swift:87]: Ah, I missed (b) down below.

It looks like that check is detecting whether the tap is in the cell—should i be checking that it's in the swipe action space? That way tapping on the other cell content would dismiss it?

**Review by n8chur — APPROVED** [review]

File: ListableUI/Sources/ListView/ListView.swift (hunk 2 above)
Comment (kyleve) [reply ListableUI/Sources/ListView/ListView.swift:87]: Ah yeah, I think I can do that, and it looks to be how UIKit works as well

Comment (kyleve) [reply ListableUI/Sources/ListView/ListView.swift:87]: Done

**Pull request merged**